Source/Dafny/DafnyMain.cs

```diff
@@ -221,15 +221,102 @@ private class IncludeComparer : IComparer<Include> {
       try {
         int errorCount = Dafny.Parser.Parse(dafnyFile.UseStdin, dafnyFile.SourceFileName, include, module, builtIns, errs, verifyThisFile, compileThisFile);
         if (errorCount != 0) {
-          return string.Format("{0} parse errors detected in {1}", errorCount, fn);
+          return $"{errorCount} parse errors detected in {fn}";
         }
       } catch (IOException e) {
-        Bpl.IToken tok = include == null ? Bpl.Token.NoToken : include.tok;
+        IToken tok = include == null ? Token.NoToken : include.tok;
         errs.SemErr(tok, "Unable to open included file");
-        return string.Format("Error opening file \"{0}\": {1}", fn, e.Message);
+        return $"Error opening file \"{fn}\": {e.Message}";
       }
       return null; // Success
     }
 
+    public static bool BoogieOnce(string baseFile, string moduleName, Microsoft.Boogie.Program boogieProgram, string programId,
```

Comment:
Line 234 until 320 were moved out of `Source/DafnyDriver/DafnyDriver.cs` to this assembly so they're accessible from the unit tests.